File: DynamicKey/AgoraDynamicKey/dart/sample/convoai_token_builder_sample.dart
import 'dart:io';

import 'package:agora_token_generator/agora_token_generator.dart';

void main() {
  final appId = Platform.environment['AGORA_APP_ID'];
  final appCertificate = Platform.environment['AGORA_APP_CERTIFICATE'];

  const channelName = 'convoai-channel';
  const rtcAccount = 'convoai-rtc-user';
  const rtcRole = RtcRole.publisher;
  const rtcTokenExpireSeconds = 3600;
  const joinChannelPrivilegeExpire = 3600;
  const publishAudioPrivilegeExpire = 3600;
  const publishVideoPrivilegeExpire = 3600;
  const publishDataPrivilegeExpire = 3600;
  const rtmUserId = 'convoai-rtm-user';
  const rtmTokenExpireSeconds = 3600;

  print('App Id: $appId');
  if (appId == null || appCertificate == null) {
    print('Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE');
    return;
  }

  final token = ConvoAITokenBuilder.buildToken(
    appId: appId,
    appCertificate: appCertificate,
    channelName: channelName,
    rtcAccount: rtcAccount,
    rtcRole: rtcRole,
    rtcTokenExpireSeconds: rtcTokenExpireSeconds,
    joinChannelPrivilegeExpire: joinChannelPrivilegeExpire,
    publishAudioPrivilegeExpire: publishAudioPrivilegeExpire,
    publishVideoPrivilegeExpire: publishVideoPrivilegeExpire,
    publishDataPrivilegeExpire: publishDataPrivilegeExpire,
    rtmUserId: rtmUserId,
    rtmTokenExpireSeconds: rtmTokenExpireSeconds,
  );
  print('ConvoAI token: $token');
}

File: DynamicKey/AgoraDynamicKey/cpp/sample/ConvoAITokenBuilderSample.cpp
// Copyright (c) 2014-2026 Agora.io, Inc.

#include <cstdlib>
#include <iostream>

#include "../src/ConvoAITokenBuilder.h"

using namespace agora::tools;

int main() {
  const char *env_app_id = getenv("AGORA_APP_ID");
  std::string app_id = env_app_id ? env_app_id : "";
  const char *env_app_certificate = getenv("AGORA_APP_CERTIFICATE");
  std::string app_certificate = env_app_certificate ? env_app_certificate : "";

  std::cout << "App Id:" << app_id << std::endl;
  if (app_id.empty() || app_certificate.empty()) {
    std::cout << "Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE" << std::endl;
    return -1;
  }

  std::string token = ConvoAITokenBuilder::BuildToken(
      app_id, app_certificate, "7d72365eb983485397e3e3f9d460bdda", "2882341273", UserRole::kRolePublisher,
      3600, 3600, 3600, 3600, 3600, "2882341273", 3600);
  std::cout << "ConvoAI token:" << token << std::endl;
  return 0;
}
